### DISCUSSION TITLE
Usage in combination with remix-auth

### DISCUSSION THREAD

**Discussion by jadlr** [airjp73/rvf#186] [Q&A]
I'm using remix-validated-form (which is great, btw :smile:) together with [remix-auth](https://github.com/sergiodxa/remix-auth) and I'm wondering If I can somehow show auth errors on fields of a `ValidatedForm`. remix-auth stores authentication errors in the session and you can then read them in a loader like so:

```typescript
export const loader: LoaderFunction = async ({ request }) => {
  await authenticator.isAuthenticated(request, {
    successRedirect: "/dashboard",
  });
  let session = await getSession(request.headers.get("cookie"));
  let error = session.get(authenticator.sessionErrorKey);
  return json({ error });
};
```

I tried using `return validationError();` but this doesn't seem to work when returning from a loader instead of an action.

Is anyone else using these libs together and found a solution to this?

Thanks!

**Comment by airjp73** [comment]
What kind of auth errors would be displayed? Entering the wrong password seems like the obvious one, but I would normally expect that to appear as some kind of top-level alert in the form rather than on the password field itself.

**Comment by Jarrodsz** [comment]
> I'm using remix-validated-form (which is great, btw :smile:) together with [remix-auth](https://github.com/sergiodxa/remix-auth) and I'm wondering If I can somehow show auth errors on fields of a `ValidatedForm`. remix-auth stores authentication errors in the session and you can then read them in a loader like so:
> 
> 
> 
> ```typescript
> 
> export const loader: LoaderFunction = async ({ request }) => {
> 
>   await authenticator.isAuthenticated(request, {
> 
>     successRedirect: "/dashboard",
> 
>   });
> 
>   let session = await getSession(request.headers.get("cookie"));
> 
>   let error = session.get(authenticator.sessionErrorKey);
> 
>   return json({ error });
> 
> };
> 
> ```
> 
> 
> 
> I tried using `return validationError();` but this doesn't seem to work when returning from a loader instead of an action.
> 
> 
> 
> Is anyone else using these libs together and found a solution to this?
> 
> 
> 
> Thanks!
> 
> 
> 
> 

Did you found a way in meantime?
